services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceDelegateMode.java

```diff
@@ -349,41 +363,130 @@ public <H, P extends ResourcePath<K>, K, R> Scatter<H, P, K> scatter(
     return finalScatter;
   }
 
-  // Select host with the least pending queue depth.
-  private <H> H selectLeastLoadedHost(List<H> hosts, VenicePath path) throws RouterException {
-    H host;
-    long minCount = Long.MAX_VALUE;
+  /**
+   * Select the least loaded host from the available healthy replicas.
+   * Behavior depends on the latencyBasedRoutingEnabled config flag:
+   *
+   * When enabled, uses average response latency as the criterion with a 1.5x spectrum
+   * threshold among the healthy host list. Ref {@link #AVG_LATENCY_SPECTRUM_FOR_HOST_SELECTION}.
+   * Randomly selects within the latency spectrum to avoid skewed traffic.
+   *
+   * When disabled (default), uses pending request count to decide on the host selection
+   * among the healthy host list.
+   *
+   * Note: Healthy host list is based on pending queue count: Ref {@link VeniceHostHealth#isHostHealthy}
```

Comment:
NIT: Healthy host list is actually based on many signals (pending request count, live instance monitor, host readiness, etc.) as you mentioned in the `Ref {@link VeniceHostHealth#isHostHealthy}`.  Let's not include these potentially incorrect/incomplete statements especially it can impact AI tooling analysis and accuracy.

Reply:
Good point, I wanted to emphasis that we are not moving away from pending queue completely. But Removed it to only refer to the VeniceHostHealth#isHostHealthy to not be misleading.